demisto_sdk/commands/validate/tests/BC_validators_test.py

```diff
@@ -891,3 +897,138 @@ def test_IsContextPathChangedValidator_remove_command():
         f"Command {command_name} has been removed from the integration. This is a breaking change, and is not allowed."
         in errors[0].message
     )
+
+
+def setup_integration_with_3_commands_and_5_args():
+    # Create a new content item with 3 commands with unique names. all commands have only 1 argument except the third command which has 2 arguments.
+    integration = create_integration_object()
+    integration.commands.append(create_integration_object().commands[0])
+    integration.commands.append(create_integration_object().commands[0])
+    integration.commands[2].args.append(create_integration_object().commands[0].args[0])
+    integration.commands[0].name = "test-command_0"
+    integration.commands[1].name = "test-command_1"
+    integration.commands[2].name = "test-command_2"
+    return integration
+
+
+def test_HaveCommandsOrArgsNameChangedValidator__fails():
+    """
+    Given
+        - A new content item with 3 commands. all commands have only 1 argument except the third command which has 2 arguments.
+        - An old content item with the same structure as the new content item, but with different command and argument names.
+    When
+        - Calling the HaveCommandsOrArgsNameChangedValidator.
+    Then
+        - Make sure the validation fails and the right error message is returned notifying only on changes in the second command name, and third command argument name.
+            - The first command has no changes.
+            - The second command name has changed from 'test-command_1' to 'new-command_1', and its arg name has changed from 'old_arg_1' to 'new_arg_1'.
+                (the args change souled be ignored since we reported on its command name change.)
+            - The third command name has not changed, but one of its arguments name has changed from 'old_arg_2' to 'new_arg_2'.
+
+    """
+    # Setup new content item with changes in command and argument names
+    new_content_item = setup_integration_with_3_commands_and_5_args()
+    new_content_item.commands[1].name = "new-command_1"
+    new_content_item.commands[1].args[0].name = "new_arg_1"
+    new_content_item.commands[2].args[1].name = "new_arg_2"
+
+    # Setup old content item with original command and argument names
+    old_content_item = setup_integration_with_3_commands_and_5_args()
+    old_content_item.commands[1].name = "old-command_1"
+    old_content_item.commands[1].args[0].name = "old_arg_1"
+    old_content_item.commands[2].args[1].name = "old_arg_2"
```

Comment:
It's done wrong here, there's a better way to init an integration with 3 commands.
see 'test_IsContainingMultipleDefaultArgsValidator_is_valid' in IN_Validators as a reference.
You can init the old content object and then use copy.deepcopy to create the new object and alter it.

Reply:
done